### PR TITLE
add alloy-metrics

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/roadmap/issues/3522

Add Alloy as alloy-metrics app in order to use it as a monitoring agent instead of Prometheus Agent.

### CODE REVIEW

File: helm/observability-bundle/values.yaml [hunk 1]
```diff
@@ -65,6 +65,26 @@ apps:
...
+    namespace: kube-system
+    # used by renovate
+    # repo: giantswarm/alloy-app
+    version: 0.3.1
+    # a list of extraConfigs for the App,
+    # It can be secret or configmap
+    # https://github.com/giantswarm/rfc/tree/main/multi-layer-app-config#example
+    extraConfigs:
+      - kind: secret
+        name: "{{ $.Values.clusterID }}-logging-secret"
```
Comment (QuentinBisson) [inline helm/observability-bundle/values.yaml:82]: Secret and configmap names are wrong here

Comment (TheoBrigitte) [reply helm/observability-bundle/values.yaml:82]: Renamed them to `$clusterID-monitoring`

Comment (QuentinBisson) [reply helm/observability-bundle/values.yaml:82]: Why not follow the other schème clusterID-monitoring-config|secret?

File: helm/observability-bundle/values.yaml [hunk 2]
```diff
@@ -65,6 +65,26 @@ apps:
...
+    namespace: kube-system
+    # used by renovate
+    # repo: giantswarm/alloy-app
+    version: 0.3.1
+    # a list of extraConfigs for the App,
+    # It can be secret or configmap
+    # https://github.com/giantswarm/rfc/tree/main/multi-layer-app-config#example
+    extraConfigs:
+      - kind: secret
+        name: "{{ $.Values.clusterID }}-monitoring-secret"
```
Comment (QuentinBisson) [inline helm/observability-bundle/values.yaml:82]: Do you want to name those alloy-monitoring-secret and config then?

Comment (TheoBrigitte) [reply helm/observability-bundle/values.yaml:82]: I would like to stick with the same naming pattern we used for logging.

Comment (QuentinBisson) [reply helm/observability-bundle/values.yaml:82]: Sure

**Review by QuentinBisson — APPROVED** [review]

File: helm/observability-bundle/values.yaml [hunk 3]
```diff
@@ -78,9 +78,6 @@ apps:
     # It can be secret or configmap
     # https://github.com/giantswarm/rfc/tree/main/multi-layer-app-config#example
     extraConfigs:
```
Comment (QuentinBisson) [inline helm/observability-bundle/values.yaml:80]: How will you set the remote write api key secret without a secret?

Comment (TheoBrigitte) [reply helm/observability-bundle/values.yaml:80]: They will be set via env vars from within another secret mount directly into Alloy pod. This secret is then not used by the app but by the pod directly.

Comment (QuentinBisson) [reply helm/observability-bundle/values.yaml:80]: That looks weird. How is that secret created on thé WC?

Comment (TheoBrigitte) [reply helm/observability-bundle/values.yaml:80]: I found no other mechanism to pass secret values to Alloy, the only possible current configuration is a single configmap.
Good catch, this secret is not created on WC, I will need to change this to have the secret created by the app then (and port this to logging-operator too for Alloy logs)

Comment (QuentinBisson) [reply helm/observability-bundle/values.yaml:80]: I think secret management is build in the grafana-agent app, either via env or something else. Maybe you Can check that?

Comment (TheoBrigitte) [reply helm/observability-bundle/values.yaml:80]: I added a Secret resource into our helm chart which allow to pass secret values via helm values.

File: helm/observability-bundle/values.yaml [hunk 4]
```diff
@@ -65,6 +65,26 @@ apps:
...
 
+  alloyMetrics:
+    appName: alloy-metrics
+    chartName: alloy
+    catalog: giantswarm-test
+    enabled: false
+    namespace: kube-system
+    # used by renovate
+    # repo: giantswarm/alloy-app
+    version: 0.3.1-0ea7e6c9fea5fa4ce5aede8f9e20ea21448fe0fb
```
Comment (QuentinBisson) [inline helm/observability-bundle/values.yaml:76]: I assume this version will have to be changed :D

Comment (TheoBrigitte) [reply helm/observability-bundle/values.yaml]: Updated to v0.4.0

**Pull request merged**